dhis2/src/main/java/org/motechproject/dhis2/rest/service/impl/DhisWebServiceImpl.java

```diff
@@ -411,7 +413,14 @@ private DhisStatusResponse createEntity(Settings settings, String uri, String js
         DhisStatusResponse status;
 
         try (InputStream content = getContentForResponse(response)) {
-            status = new ObjectMapper().readValue(content, DhisStatusResponse.class);
+            String contentString = IOUtils.toString(content);
+            status = new ObjectMapper().readValue(contentString, DhisStatusResponse.class);
+            if (status.getStatus() == DhisStatus.ERROR) {
+                LOGGER.debug(String.format("DHIS2 status response error details: %s", contentString));
+                String msg = String.format("Error in DHIS2 status response.");
+                statusMessageService.warn(msg, MODULE_NAME);
+                throw new DhisWebException(msg);
```

Comment:
1. From JIRA 'Ideally task should fail properly with an error'. I see that task fails with exception and brief error message. We should extend this to fail properly with an error
2. Additionally as far as I understand, warn method from statusMessageService is logging this message. So now we are logging on debug level using LOGGER and then using service ? We should standardize this